### DISCUSSION TITLE
Hide Username Field & Setting from Checkout Blocks

### DISCUSSION THREAD

**Discussion by dominique-vias** [woocommerce/woocommerce#49023] [Checkout Flow]
## Overview

As part of our ongoing improvements to the Checkout Blocks, we are updating the account creation process to make it more intuitive and streamlined for merchants and shoppers alike. One significant enhancement is hiding the username creation step during checkout.

This update aims to simplify the account creation flow, reduce friction during checkout, and provide a more user-friendly experience that encourages customers to create accounts.

![Alt text](https://github.com/woocommerce/woocommerce/assets/99638528/c3173561-1bf2-4dc3-b4ce-5e6b1535a5f5)
*Example mockup*

> [!NOTE]  
> This change only affects the Checkout Blocks and do not impact the classic checkout experience.

<br>

## Background

In the Accounts & Privacy settings in Woo Admin, there is an option that can introduce friction during the checkout flow, whether the merchant chooses to enable it or not:

<br>

- [ ] `When creating an account, automatically generate an account username for the customer based on their name, surname or email`

![Alt text](https://github.com/woocommerce/woocommerce/assets/99638528/a91b8a18-55cc-4d5a-971e-ba4c7ff09528)
*Current Accounts & Privacy settings*

### Auto-Generating Usernames
Automatically generating usernames based on the customer's name, surname, or email is an outdated practice and can result in usernames that users don’t prefer or are difficult to remember.

### Manual Username Creation
When the auto-generation of usernames is turned off, shoppers are required to manually create usernames. Including a username creation field can complicate the checkout process. This is true whether creating an account for authenticated checkouts or when guest shoppers opt in to create an account. It adds complexity and is perceived as unrelated to completing their purchase efficiently. This can lead to higher abandonment rates and potential confusion for customers during a critical phase of their shopping experience.
<br>

## The Solution
To ensure a frictionless experience for shoppers and encourage more customers to create accounts, we are `hiding the option to create usernames` during the checkout process and instead use the shopper's email as the primary login. This shift will:
<br>
- Simplify the account creation process by reducing the number of steps and decisions required during checkout
- Be a good step towards aligning with standard account creation and management practices
- Encourage account creation by providing a smoother, less intrusive process

<br>

To implement this, we will make some adjustments to the admin settings panel available for merchants  (along with other minor improvements as seen below).

**Specifically, we will hide the option to create usernames from the Accounts & Privacy settings.** We will still to auto-generate usernames internally based on the email and/or other name fields. Usernames for existing accounts can still be used to log in and will appear in places where they are currently used, such as admin panels and customer management interfaces.

<br>

![Alt text](https://github.com/woocommerce/woocommerce/assets/99638528/d3e2dff9-9e40-4cd4-95f1-460a93198a38)
*IA improvements and hiding username creation*

We can reintroduce this setting if we uncover strong cases to do so. This change will impact any store using the Checkout blocks. Users of the classic checkout experience won't be impacted by these changes. 
<br>

## Next Steps

We will continue to analyze the shopper journey and continue to improve the account creation flow where it involves checkout to further streamline the experience.

**Comment by senadir** [comment]
Would we continue by hiding any mentions of username in customer emails and my account screen?

**Reply by dominique-vias** [reply]
> Would we continue by hiding any mentions of username in customer emails and my account screen?

Email— I would be inclined to hide because the post-checkout email is essentially a summary of what has transpired in checkout, which doesn't include the username...

My Account— Hmmm I'd say they should have the ability to change the display name...
